paramiko/ecdsakey.py

```diff
@@ -89,29 +89,35 @@ def get_by_key_length(self, key_length):
                 return curve
 
 
+@register_pkey_type
```

Comment:
I think this is OK, but here's an alternate, more direct alternative: after defining the class, call this at the end of the source file:

```python
class ECDSAKey(PKey):
    ...

register_pkey_type(ECDSAKey, legacy_type="EC", openssh_type_prefix="ecdsa-sha2-")
```

just an idea

Reply:
I prefer the decorator myself. Something at the very end of a file is likely to be missed.

I could take the `legacy_type` and `openssh_type_prefix` as argument to the decorator, but since they're also accessed from within the class itself (in `_from_private_key`), I prefer them as explicit attributes.

Reply:
I see your point about keeping the class members `LEGACY_TYPE` etc, I didn't notice the other usage.